Documentation/guides/profiling.md

```diff
@@ -0,0 +1,244 @@
+    09-04 14:44:12.238  9126  9126 I monodroid-timing: TypeManager.cctor start: 1567626252195.41
+    09-04 14:44:12.255  9126  9126 I monodroid-timing: TypeManager.cctor time: 1567626252254.66 [elapsed: 59.248 ms]
+    09-04 14:44:12.255  9126  9126 I monodroid-timing: JNIEnv.Initialize: TypeManager init time: elapsed 63 ms]
+    09-04 14:44:12.363  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives ("Android.Runtime.UncaughtExceptionHandler, Mono.Android", 0x7fd311c2cc) start: 1567626252360.4
+    09-04 14:44:12.420  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives total time: 1567626252420.89 [elapsed: 60.492 ms]
+    09-04 14:44:12.421  9126  9126 I monodroid-timing: Runtime.register: end time; elapsed: 0s:62::781621
+    09-04 14:44:12.437  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives ("Java.Interop.TypeManager+JavaTypeManager, Mono.Android, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null", 0x7fd311b9ec) start: 1567626252437.71
+    09-04 14:44:12.439  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives total time: 1567626252439.4 [elapsed: 1.684 ms]
+    09-04 14:44:12.439  9126  9126 I monodroid-timing: Runtime.register: end time; elapsed: 0s:1::757865
+    09-04 14:44:12.446  9126  9126 I monodroid-timing: JNIEnv.Initialize end: elapsed 479 ms
+    09-04 14:44:12.446  9126  9126 I monodroid-timing: Runtime.init: end native-to-managed transition; elapsed: 0s:443::816399
+    09-04 14:44:12.446  9126  9126 I monodroid-timing: Runtime.init: end, total time; elapsed: 0s:480::702704
+    09-04 14:44:12.459  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives ("HelloWorld.MainActivity, HelloWorld", 0x7fd311ceec) start: 1567626252459.23
+    09-04 14:44:12.470  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives total time: 1567626252470.37 [elapsed: 11.144 ms]
+    09-04 14:44:12.470  9126  9126 I monodroid-timing: Runtime.register: end time; elapsed: 0s:11::259949
+    09-04 14:44:12.559  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives ("Android.Views.View+IOnClickListenerImplementor, Mono.Android", 0x7fd311b5fc) start: 1567626252559.64
+    09-04 14:44:12.561  9126  9126 I monodroid-timing: JNIEnv.RegisterJniNatives total time: 1567626252561.82 [elapsed: 2.182 ms]
+    09-04 14:44:12.561  9126  9126 I monodroid-timing: Runtime.register: end time; elapsed: 0s:2::244636
+
+These are timings taken at different points during Xamarin.Android
+startup: across native and managed code.
+
+An example of these log messages would be [this managed
+call][managed_timing] or [this native call][native_timing].
+
+[managed_timing]: https://github.com/xamarin/xamarin-android/blob/faf2a3d7271e2e321b2fa500f85fa0f824abcf89/src/Mono.Android/Android.Runtime/JNIEnv.cs#L160
+[native_timing]: https://github.com/xamarin/xamarin-android/blob/6be4bdbcfd9f2fbe6267aceb934394bcdd0f13b4/src/monodroid/jni/monodroid-glue.cc#L2062
+
```

Comment:
It is also possible to process this logcat output like described here:

https://github.com/xamarin/xamarin-android/commit/6436b11fdadf305d6c3b62fd4744dae839929310

It is only available in the build tree though. Maybe we can make it available in the installed XA as well?

There's also PR related to this, which I should revive, https://github.com/xamarin/xamarin-android/pull/3369

The output looks like this:
```
Task "ProcessLogcatTiming"
  Time:      0ms process start, application: '' PID: 3011
  Time:     57ms Message: Runtime.init: Mono runtime init; elapsed: 0s:0::44834
  Time:     65ms Message: Finished loading assemblies: preloaded 1 assemblies; wasted time: 0s:0::49
  Time:     73ms Message: JNIEnv.Initialize start
  Time:     78ms Message: JNIEnv.Initialize: Logger JIT/etc. time: elapsed 3 ms]
  Time:    101ms Message: JNIEnv.Initialize: managed runtime init time: elapsed 22 ms]
  Time:    113ms Message: TypeManager.cctor start: 1567669218343.18
  Time:    116ms Message: TypeManager.cctor time: 1567669218357.74 [elapsed: 14.563 ms]
  Time:    117ms Message: JNIEnv.Initialize: TypeManager init time: elapsed 15 ms]
  Time:    156ms Message: JNIEnv.RegisterJniNatives ("Android.Runtime.UncaughtExceptionHandler, Mono.Android", 0xffe3c454) start: 1567669218396.28
  Time:    163ms Message: JNIEnv.RegisterJniNatives total time: 1567669218404.38 [elapsed: 8.097 ms]
  Time:    163ms Message: Runtime.register: end time; elapsed: 0s:8::800097
  Time:    164ms Message: JNIEnv.RegisterJniNatives ("Java.Interop.TypeManager+JavaTypeManager, Mono.Android, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null", 0xffe3bcf4) start: 1567669218405.08
  Time:    164ms Message: JNIEnv.RegisterJniNatives total time: 1567669218405.86 [elapsed: 0.787 ms]
  Time:    164ms Message: Runtime.register: end time; elapsed: 0s:0::813062
  Time:    166ms Message: JNIEnv.Initialize end: elapsed 105 ms
  Time:    166ms Message: JNIEnv.Initialize end: elapsed 105 ms
  Time:    166ms Message: Runtime.init: end native-to-managed transition; elapsed: 0s:100::17520
  Time:    166ms Message: Runtime.init: end native-to-managed transition; elapsed: 0s:100::17520
  Time:    166ms Message: Runtime.init: end, total time; elapsed: 0s:109::900001
  Time:    173ms Message: JNIEnv.RegisterJniNatives ("xatemplateaot.MainActivity, xatemplateaot", 0xffe3cbe4) start: 1567669218414.33
  Time:    173ms Message: JNIEnv.RegisterJniNatives total time: 1567669218414.89 [elapsed: 0.552 ms]
  Time:    173ms Message: Runtime.register: end time; elapsed: 0s:0::594434
  Time:    232ms Message: JNIEnv.RegisterJniNatives ("Android.Views.View+IOnClickListenerImplementor, Mono.Android", 0xffe3bb54) start: 1567669218473.37
  Time:    232ms Message: JNIEnv.RegisterJniNatives total time: 1567669218473.59 [elapsed: 0.219 ms]
  Time:    232ms Message: Runtime.register: end time; elapsed: 0s:0::241370
   -- Performance summary --
  Last timing message: 232ms
Done executing task "ProcessLogcatTiming".
```